src/back/TicketManager/src/Core/TicketManager.Core.Domain/Events/Sector.cs

```diff
@@ -1,6 +1,33 @@
+using TicketManager.Core.Domain.Common;
+
 namespace TicketManager.Core.Domain.Events;
 
-public record Sector(Guid EventId, string Name, int PriceInSmallestUnit, int NumberOfColumns, int NumberOfRows)
+public class Sector: 
+    IAggregateRoot<Guid>, IOptimisticConcurrent
 {
-    public (Guid EventId, string Name) Id => (EventId, Name);
+    public Guid Id { get; private init; }
+    public Guid EventId { get; private init; }
+    public string Name { get; private set; } = null!;
+    public int PriceInSmallestUnit { get; private set; }
+    public int NumberOfColumns { get; private set; }
+    public int NumberOfRows { get; private set; }
+    public int NumberOfSeats => NumberOfRows * NumberOfColumns;
+    private List<SeatReservation> seatReservations = new();
+    public IReadOnlyCollection<SeatReservation> SeatReservations => seatReservations.AsReadOnly();
+    public DateTime DateModified { get; set; }
+    
+    public Sector(Guid eventId, string name, int priceInSmallestUnit, int numberOfColumns, int numberOfRows)
+    {
+        EventId = eventId;
+        Name = name;
+        PriceInSmallestUnit = priceInSmallestUnit;
+        NumberOfColumns = numberOfColumns;
+        NumberOfRows = numberOfRows;
+    }
```

Comment:
Set `Id`.
Also add parameterless private constructor.

Reply:
Funny thing, apparently `Guid`s are really initialized with `Guid.NewGuid()` and not `Guid.Empty`. I have proper entries in database so I didn't really catch that mistake